Create backbone of application.

Niedzwiedz opened the pull request:
Create models for Game Room Application. Models include: User, TeamUser, Team, Match and Game.
Add basic functionality, validations and test of models.  Populate seeds.rb and add CarrierWave support.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/niedzwiedz/montekombat/1)

<!-- Reviewable:end -->

Niedzwiedz committed:
Remove unstaged files

Niedzwiedz committed:
Tests
Add game test

Niedzwiedz committed:
[fix] Remove file \

Niedzwiedz committed:
[fix] Add password to user

Niedzwiedz committed:
Fixed some issues and added moar tests

Niedzwiedz committed:
Add more useful tests fixed style with rubocop

Added tests and functionality to prevent player from being in two teams
that are currently in match and to be duplicated in one team

Niedzwiedz committed:
Change sqlite3 to postgres, add some test and seed

Changed database, added tests and functionalities for match type,
match status and user account type. Filled seeds.rb.

Niedzwiedz committed:
Add CarrierWave support

Added CarrierWave support, fixed situtation with storing lots of images
after tests.

monttda commented on app/models/match.rb:
```diff
@@ -1,5 +1,36 @@
+
+  validates :points_for_team1, presence: true
+  validates :points_for_team2, presence: true
+  validates :team_1, presence: true
```
group all  the attributes that have the validation of presence into one validation command only

monttda commented on app/models/match.rb:
```diff
@@ -1,5 +1,36 @@
+  private
+
+  def same_player_cant_be_in_both_teams
+    same_players_check if !team_1.nil? && !team_2.nil?
```
use present? instead of negating nil?

monttda commented on spec/models/game_spec.rb:
```diff
@@ -0,0 +1,22 @@
+  let(:game) { build(:game) }
+  context "When its valid" do
+    it "has name" do
+      byebug
```
remove this bye bug

monttda commented on spec/models/game_spec.rb:
```diff
@@ -0,0 +1,22 @@
+      expect(game.name).to be_present
+    end
+
+    it { expect(game.game_picture ).not_to be_present }
```
is this test green ?

monttda commented on spec/models/match_spec.rb:
```diff
@@ -0,0 +1,75 @@
+                  points_for_team2: 2)
+  end
+  let(:user) { create(:user) }
+  context "is valid" do
```
"is valid when" so when the test output text appears it says "is valid when it .."

monttda commented on spec/models/match_spec.rb:
```diff
@@ -0,0 +1,75 @@
+
+    it "has team_2 points" do
+      expect(match.points_for_team2).to eq(2)
+    end
```
most of this tests are not needed since all you need is one test in which when you have all the right attributes it is valid. The  ones that must exist are the not valid for each missing attribute

monttda commented on spec/models/match_spec.rb:
```diff
@@ -0,0 +1,75 @@
+      match.points_for_team1 = nil
+      expect(match).not_to be_valid
+    end
+    it "doesn't have team_2 points" do
```
use context for the description of the test and then inside to do all the operations use a before block or a let  so you only have the expectation inside the `it`. This applies for every tests file

Niedzwiedz committed:
First code review fix

Clean/repair some tests, validations. Removed unnesessary tests.

monttda commented on db/migrate/20160905122239_create_team_user.rb:
```diff
@@ -1,8 +1,8 @@
-      t.belongs_to :team, index: true
-      t.belongs_to :user, index: true
+      t.belongs_to :team, index: true, unique: true
+      t.belongs_to :user, index: true, unique: true
```
this is not a composite index :p it will not work, you need to explicitly create an index `add_index :users, [:email, :account_id], unique: true` something like that

monttda commented on spec/models/game_spec.rb:
```diff
@@ -2,21 +2,22 @@
+    end
   end
-  context "When its invalid" do
+  context "When it's invalid" do
```
more like "with invalid attributes" and the other "with valid attributes" this because you don't describe the result on  the text but the conditions that make X result happen. The result is what you write on the expect ( in this case not being valid

monttda commented on spec/models/match_spec.rb:
```diff
@@ -1,75 +1,71 @@
-
-    it "does have team_1" do
-      expect(match.team_1).to be_present
+  context "When it's valid" do
```
same a previous file

monttda commented on spec/models/match_spec.rb:
```diff
@@ -1,75 +1,71 @@
-      expect(match.team_1).to be_present
+  context "When it's valid" do
+    it "it's attributes are present and correct" do
+      expect(match).to be_valid
```
no need for this it description since the context will say this so you only need the `it '' do ... end`

monttda commented on spec/models/match_spec.rb:
```diff
@@ -1,75 +1,71 @@
-    it "doesn't have team_1" do
-      match.team_1 = nil
-      expect(match).not_to be_valid
+    context "team 2 doesn't exist" do
```
is not present instead of doesn't exist

monttda commented on spec/models/team_spec.rb:
```diff
@@ -3,34 +3,42 @@
   let(:team) { build(:team, name: "somename") }
   let(:user) { build(:user, username: "somename") }
-  context "when it's valid" do
+  context "When it's valid" do
```
same a the previous file

monttda commented on spec/models/user_spec.rb:
```diff
@@ -4,73 +4,73 @@
-    it "has email" do
-      expect(user.email).to be_present
-    end
-    it "has username" do
```
same as previous files

monttda commented on spec/models/user_spec.rb:
```diff
@@ -4,73 +4,73 @@
+      before do
+        user.teams << create(:team)
+        user.teams << create(:team)
+      end
```
use a `create_list` instead  of assigning twice

Niedzwiedz committed:
Second code review fix

Even better specs, create list.
TODO: validate that no empty team can be in match

Niedzwiedz committed:
Add empty team in match validation

move same team in multiple matches validation to match rspec file
TODO: rspec file for match needs clean up

Niedzwiedz committed:
Clean match test file

jotka87 commented:
Reviewed 98 of 105 files at r1, 2 of 8 files at r2, 6 of 6 files at r3.
Review status: all files reviewed at latest revision, 33 unresolved discussions.

---

_[app/models/.DS_Store, line 0 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTdP98WOMuXdqAedEE:-KRTdP98WOMuXdqAedEF:b8zyck5) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/.DS_Store#L0)):_
Please add that file to the `.gitignore` file.

---

_[app/models/\, line 0 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTdXBPXpJc2f6AGAiC:-KRTdXBPXpJc2f6AGAiD:b-x667di) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/#L0)):_
What's that? Please remove it.

---

_[app/models/game.rb, line 1 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTe2UIiLQbeNMzdgqc:-KRTe2UIiLQbeNMzdgqd:b-xmrd8q) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/game.rb#L1)):_

> ``` Ruby
> class Game < ActiveRecord::Base
> ```

Why this change? http://blog.bigbinary.com/2015/12/28/application-record-in-rails-5.html

---

_[app/models/match.rb, line 21 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRThKEuL0IzyMasAbFw:-KRThKEvThKFbGpQawym:bob0hc2) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/match.rb#L21)):_

> ``` Ruby
>     finished: 2,
>   }
>   validates :status, inclusion: { in: %w(upcoming in_progress finished) }
> ```

you don't need to validate names, when incorrect key is passed, an `ArgumentError` will be raised (unless there were some changes in Rails 5).

---

_[app/models/match.rb, line 31 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTfQ34qoHBxZuhE7Wb:-KRTfQ34qoHBxZuhE7Wc:b-vynz9v) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/match.rb#L31)):_

> ``` Ruby
> 
>   def same_players_check
>     unless team_1.users & team_2.users == []
> ```

Use ids collection instead of objects collection - `team_1.user_ids & team_2.user_ids`. 

---

_[app/models/match.rb, line 42 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTeXwmtFHCM1xwwyKW:-KRTeXwmtFHCM1xwwyKX:b-30ig0k) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/match.rb#L42)):_

> ``` Ruby
> 
>   def empty_teams_check
>     errors[:team] << "There is an empty team" if team_1.users.empty? || team_2.users.empty?
> ```

That validation should go to the `Team` model.

---

_[app/models/team.rb, line 4 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRThj5x5BjPLavYMRgr:-KRThj5yVQOJFhsSaZPl:b-hpgn6a) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/team.rb#L4)):_

> ``` Ruby
>   has_many :matches
>   has_many :users, through: :team_user
>   has_many :team_user
> ```

plural form for `has_many` associations - `team_users`, in `through` also.

---

_[app/models/team_user.rb, line 4 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTkmHBkaqP07uFlYuc:-KRTkmHCBGQcmNpbyBVN:b-fatcvf) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/team_user.rb#L4)):_

> ``` Ruby
>   belongs_to :team
>   belongs_to :user
>   validates :team, uniqueness: { scope: :user }
> ```

`team` and `user` should be required.

---

_[app/models/user.rb, line 2 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTkv548aB6jsDMTM2b:-KRTkv55nC30iGhr7FmD:b-16p469) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/user.rb#L2)):_

> ``` Ruby
> class User < ApplicationRecord
>   has_many :teams, through: :team_user
> ```

`team_users`

---

_[app/models/user.rb, line 3 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTkwtqAybIFxHjSb7J:-KRTkwtqAybIFxHjSb7K:b-16p469) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/user.rb#L3)):_

> ``` Ruby
> class User < ApplicationRecord
>   has_many :teams, through: :team_user
>   has_many :team_user
> ```

`team_users`

---

_[app/models/user.rb, line 16 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTlDYLezCxyUfK8UrL:-KRTlDYLezCxyUfK8UrM:bwycrzw) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/user.rb#L16)):_

> ``` Ruby
>     admin: 1,
>   }
>   validates :account_type, inclusion: { in: %w(normal admin) }
> ```

same comment like in `Match` model

---

_[db/migrate/20160905121618_create_user.rb, line 4 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTlsAlI2d6juPHG8eR:-KRTlsAmS1GVyYfDnktT:b-vu2p7l) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/db/migrate/20160905121618_create_user.rb#L4)):_

> ``` Ruby
>   def change
>     create_table :users do |t|
>       t.string :username
> ```

please add `null: false` to the required attributes. Also `username` and `email` should be uniq.

---

_[db/migrate/20160905121649_create_match.rb, line 6 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTmG75WeuIDUwGZ-4m:-KRTmG75WeuIDUwGZ-4n:b-9vhguo) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/db/migrate/20160905121649_create_match.rb#L6)):_

> ``` Ruby
>       t.belongs_to :game, index: true
>       t.belongs_to :team_1, index: true
>       t.belongs_to :team_2, index: true
> ```

add `null: false` to the required attributes.

---

_[db/migrate/20160905122239_create_team_user.rb, line 5 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTmT3hmavQslke9Lqi:-KRTmT3hmavQslke9Lqj:b-9vhguo) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/db/migrate/20160905122239_create_team_user.rb#L5)):_

> ``` Ruby
>     create_table :team_users do |t|
>       t.belongs_to :team, index: true
>       t.belongs_to :user, index: true
> ```

add `null: false` to the required attributes.

---

_[db/migrate/20160905122621_create_game.rb, line 4 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTmaOgbW3wgJAg3Hbs:-KRTmaOhwEEiWMIpLkHu:b-c1pruz) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/db/migrate/20160905122621_create_game.rb#L4)):_

> ``` Ruby
>   def change
>     create_table :games do |t|
>       t.string :name
> ```

`name` should be required

---

_[spec/models/game_spec.rb, line 7 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTqinoniPLc2S9ui3E:-KRTqinoniPLc2S9ui3F:bpu7asn) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/spec/models/game_spec.rb#L7)):_

> ``` Ruby
>   context "with valid attributes" do
>     it "has name" do
>       expect(game.name).to be_present
> ```

it doesn't really test if `name` is required, instead you should check if game without name is invalid.

---

_[spec/models/match_spec.rb, line 8 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTMbnJIM47kvM5-KYA-r2-8:-KRTreiXSg1_weYYAtEK:bd8gfaj) ([raw file](https://github.com/niedzwiedz/montekombat/blob/ad843ef884958f8b880082851d2bda03a4feb5b9/spec/models/match_spec.rb#L8)):_
<details><summary>

<i>Previously, monttda (Daniel Montenegro) wrote…</i></summary>



> no need for this it description since the context will say this so you only need the `it '' do ... end`
> </details>
> This test is redundant - the next test checks exactly the same, so I recommend to remove this one.

---

_[spec/models/team_spec.rb, line 12 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTssUMVaEt_25OziyH:-KRTssUMVaEt_25OziyI:b3gvn7t) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/spec/models/team_spec.rb#L12)):_

> ``` Ruby
>     end
>     it "has unique name" do
>       expect(team.name).to be_present
> ```

you have different thing described and different thing you're testing - if you want to check if uniqueness is checked properly, you should try to create a team with duplicated name and then check if it's invalid.

---

_[spec/models/team_user_spec.rb, line 0 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTtgkg9A6YQ59U__Oa:-KRTtgkg9A6YQ59U__Ob:bszwcok) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/spec/models/team_user_spec.rb#L0)):_
empty file

---

_[spec/models/user_spec.rb, line 30 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTuSLQlq0kY1RUO_eY:-KRTuSLQlq0kY1RUO_eZ:b2vebt7) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/spec/models/user_spec.rb#L30)):_

> ``` Ruby
>         user.username = nil
>       end
>       it { expect(user).not_to be_valid }
> ```

for this kind of tests I recommend using this gem: (but with the `expect` syntax) https://github.com/thoughtbot/shoulda-matchers#should-vs-is_expectedto

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fixes for Kamila code review #1

Niedzwiedz committed:
Delete useless comments

Niedzwiedz committed:
Add match_type and status to presence validator

Niedzwiedz committed:
Repair match model

Niedzwiedz committed:
Remove DS_Store

Niedzwiedz commented:
Review status: 86 of 105 files reviewed at latest revision, 33 unresolved discussions.

---

_[app/models/match.rb, line 31 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTfQ34qoHBxZuhE7Wb:-KRTyNxfzprdxfCJyhsB:b-qvkgw2) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/match.rb#L31)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> Use ids collection instead of objects collection - `team_1.user_ids & team_2.user_ids`. 
> </details>
> Wow! that's neat

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1)_

<!-- Sent from Reviewable.io -->

jotka87 commented:
Reviewed 21 of 21 files at r4.
Review status: all files reviewed at latest revision, 22 unresolved discussions.

---

_[Gemfile, line 59 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRYuQC867KZigqluGzk:-KRYuQC867KZigqluGzl:b-490xhb) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6268642ce27a3204abec2ca4bdaf7eb40ff8a085/Gemfile#L59)):_

> ``` Ruby
> 
> group :test do
>   gem "shoulda-matchers", "~> 3.1"
> ```

it won't be used, so it should be removed from `Gemfile`

---

_[app/models/match.rb, line 42 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTeXwmtFHCM1xwwyKW:-KRYuyUFAorvZ3f2zdWi:b-rfatr1) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/match.rb#L42)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> That validation should go to the `Team` model.
> </details>
> it's still here :)

---

_[app/models/team.rb, line 8 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRYv2QGLlxK1RKlAq7X:-KRYv2QGLlxK1RKlAq7Y:bjkkfpp) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6268642ce27a3204abec2ca4bdaf7eb40ff8a085/app/models/team.rb#L8)):_

> ``` Ruby
>   validates :name, uniqueness: true, length: { minimum: 3, maximum: 66 }
>   validate :player_cant_be_duplicated_in_team
>   validates_presence_of :users
> ```

please follow the syntax: `validates :users, presence: true`, all the validations should follow the same convention and `validate_presence_of` is the old one.

---

_[app/models/team_user.rb, line 5 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRYvO3t9d_vSMg_8tJN:-KRYvO3t9d_vSMg_8tJO:by3yhx) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6268642ce27a3204abec2ca4bdaf7eb40ff8a085/app/models/team_user.rb#L5)):_

> ``` Ruby
>   belongs_to :user
>   validates :team, uniqueness: { scope: :user }
>   validates_presence_of :team, :user
> ```

`validates :team, :user, presence: true`

---

_[db/schema.rb, line 13 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRYvlTZrRaB-MuZQaLP:-KRYvlTZrRaB-MuZQaLQ:b-cn5ub5) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6268642ce27a3204abec2ca4bdaf7eb40ff8a085/db/schema.rb#L13)):_

> ``` Ruby
> # It's strongly recommended that you check this file into your version control system.
> 
> ActiveRecord::Schema.define(version: 20160905122621) do
> ```

`schema.rb` was not updated but you have changes in migrations, please run `rake db:migrate` and commit the changed `schema.rb`

---

_[spec/rails_helper.rb, line 60 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRYwGczPw-J2oyqEyOz:-KRYwGczPw-J2oyqEyP-:bgumvpr) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6268642ce27a3204abec2ca4bdaf7eb40ff8a085/spec/rails_helper.rb#L60)):_

> ``` Ruby
> end
> 
> Shoulda::Matchers.configure do |config|
> ```

we established not to use shoulda-matchers due to lack of support for rails 5, so the changes connected with setup for it should be removed

---

_[spec/factories/team_factory.rb, line 6 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRYwgdIGSE6LESGwQvI:-KRYwgdJjRog8tesZdpj:b-ypew47) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6268642ce27a3204abec2ca4bdaf7eb40ff8a085/spec/factories/team_factory.rb#L6)):_

> ``` Ruby
> 
>     factory :team_with_users do
>       after(:build) do |team|
> ```

I would recommend to move that callback to `team` factory because in most cases you would need a proper team instance (with members assigned)

---

_[app/models/.DS_Store, line  at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTdP98WOMuXdqAedEE:-KRZ2WbWvlsCOWP9M3bc:b-wbagl2) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/.DS_Store#L)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> Please add that file to the `.gitignore` file.
> </details>
> please remove that file

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fix for Kamila code review #2

jotka87 commented:
Reviewed 10 of 10 files at r5.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fix schema and team_user migration

Niedzwiedz commented:
Review status: 102 of 105 files reviewed at latest revision, 17 unresolved discussions.

---

_[app/models/match.rb, line 42 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTeXwmtFHCM1xwwyKW:-KRZ3nMO4sFCGEcecJsL:b-2dr5o2) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/match.rb#L42)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> it's still here :)
> </details>
> oh sorry, i only removed validate with this method in case i f up :D

---

_[app/models/.DS_Store, line  at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1#-KRTdP98WOMuXdqAedEE:-KRbpf1vgI2byl5_dJv6:b-jirkf7) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c9dda7bab22596b1d9c2eb72c70f2ebaf8806f03/app/models/.DS_Store#L)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> please remove that file
> </details>
> Emmm.. i think it is removed already. I can't see it 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1)_

<!-- Sent from Reviewable.io -->

jotka87 commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fixed uniqueness in user

Niedzwiedz committed:
Fix uniqueness of user #2

jotka87 commented:
Reviewed 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/1)_

<!-- Sent from Reviewable.io -->
